example_queries/get_service_name.rs.ref

```diff
@@ -83,87 +103,123 @@ impl Filter {
     }
 
     pub fn set_whoami(&mut self) {
-        let my_node_wrapped = self
-            .filter_state
-            .get("node.metadata.WORKLOAD_NAME");
-        if my_node_wrapped.is_none() {
+        if !self.filter_state.contains_key("node.metadata.WORKLOAD_NAME") {
             print!("WARNING: filter was initialized without envoy properties and thus cannot function");
             return;
         }
-        let my_node = my_node_wrapped
-            .unwrap()
-            .string_data
-            .clone()
-            .unwrap();
+        let my_node = self
+            .filter_state["node.metadata.WORKLOAD_NAME"].clone();
         self.whoami = Some(my_node);
         assert!(self.whoami.is_some());
     }
 
-    pub fn store_headers(&mut self, x: Rpc) {
-        // store path as well as properties
-        let prop_str = format!("{uid}_properties_path", uid=x.uid);
-        if x.headers.contains_key("properties_path") {
-            if self.envoy_shared_data.contains_key(&prop_str) {
-                self.envoy_shared_data.get_mut(&prop_str).unwrap().push_str(",");
-                self.envoy_shared_data.get_mut(&prop_str).unwrap().push_str(&x.headers["properties_path"]);
-                self.envoy_shared_data.get_mut(&prop_str).unwrap().push_str(";");
-                self.envoy_shared_data.get_mut(&prop_str).unwrap().push_str(self.whoami.as_ref().unwrap());
-            }
-            else {
-                // add yourself
-                let mut cur_path = x.headers["properties_path"].clone();
-                cur_path.push_str(";");
-                cur_path.push_str(self.whoami.as_ref().unwrap());
-                self.envoy_shared_data.insert(prop_str, cur_path);
-            }
+    pub fn store_headers(&mut self, uid_64: u64, headers: IndexMap<String,String>) {
+        // If you don't have data, nothing to store
+        if !headers.contains_key("ferried_data") { 
+            print!("WARNING: no ferried data\n");
```

Comment:
Why use print! when you can use println!?

Reply:
Sometimes you just forget println exists, or at least I do haha.  But it should be a moot point now because everything has been changed to logs.